crates/re_viewer/src/ui/welcome_screen/mod.rs

```diff
@@ -0,0 +1,239 @@
+mod example_page;
+mod welcome_page;
+
+use egui::Widget;
+use re_log_types::LogMsg;
+use re_smart_channel::{ReceiveSet, SmartChannelSource};
+use re_ui::ReUi;
+use std::hash::{Hash, Hasher};
+use welcome_page::welcome_page_ui;
+
+#[derive(Debug, Default, PartialEq)]
+enum WelcomeScreenPage {
+    #[default]
+    Welcome,
+    Examples,
+}
+
+// used for the top-level scroll area ID
+impl Hash for WelcomeScreenPage {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        "welcome_screen_page".hash(state);
+        std::mem::discriminant(self).hash(state);
+    }
+}
```

Comment:
Why do you implement this manually?

Reply:
cleaned that up. who knows what the heck went through my mind...